### DISCUSSION TITLE
这个主题无比适合制作公司官网

### DISCUSSION THREAD

**Discussion by Jetive** [pengzhanbo/vuepress-theme-plume#2] [Ideas]
这几天看了不少框架，也看了不少vuepress的主题，发现这是个无敌主题。

最近在筹划公司官网，意图是以内容为主，形式为辅，所以非常希望找一套现成的方案去实现前端页面和设计。看来看去vuepress最合适，这里不展开说vuepress的优点，只举两个例子即可：

- [flomo](https://flomoapp.com)
- [cubox](https://cubox.pro)

这两个产品的官网都是用vuepress做的，尤其是flomo，几乎全部页面都是。

而对比官方主题，plume的优势是非常非常大的。plume最让我喜爱的一点就是移动端的表现。

现在的网站访问基本上都是在手机上了，就算是告诉别人自己的品牌名，对方会在手机上搜索（可能是百度app）然后打开网站，就算是告诉别人网址，也是通过微信发给对方，然后在微信里直接点开，除了极少数人群（如程序员）会在电脑上打开网页，大部分场景都已经在手机端了。

plume对比官方主题最大的改进是侧边栏在移动端的表现，使用了双层顶条，这一点其他任何主题都没能做好。为这一点，我会坚决选择plume。

plume还非常非常适合那种有公司介绍又有产品介绍的官网，比如上面提到的两个例子，都是有一个类似于使用说明书的文档，这类内容首先是vuepress的拿手好戏，另外又要有介绍页面，在介绍页面方面，plume很好地补充了官方主题的不足。

我对plume有一点点小的改进建议：

1. 移动端顶栏，双层顶条，建议可设置是否自动隐藏。说真的现在滑屏自动隐藏挺难受的。
2. 移动端一级导航栏，能否用一个写在url中的参数将其永久隐藏？如：https://aaa.com/bbb/ccc.html 正常显示，https://aaa.com/bbb/ccc.html?ddd=false 不显示一级导航栏。这样一来，就可以把一个公司旗下的多个网站整合起来了，可以整合使用，也可以分开单独使用。举例：pengzhanbo.cn整个网站可以整合起来使用，也可以把vuepress-theme-plume单独成一个站，整合使用时，浏览vuepress-theme-plume的人可以通过一级导航看博客等其他内容，单独使用时就是专门的vuepress-theme-plume文档站，不影响体验。

以上完全是个人的一点想法哈，没什么根据，只是自己有用，建议不一定合理。

明天开始我就要开始用plume建立自己的官网了，做好之后第一时间来找鹏展博大神汇报。

**Comment by pengzhanbo** [comment]
😂 谢谢夸奖。 
其实 这个主题中，关于 导航栏 部分的 样式与交互，主要是借鉴的 vitepress 的默认主题，比较符合我的期望，然后 notes 部分的样式也是 vitepress 的默认主题的效果，我主要是把它迁移到了 vuepress ，并在此基础上 对 markdown 做了增强。
然后继续在此基础上添加 功能。 我一开始的设计期望就是 集成  博客 + 笔记/文档 的功能。 在完成了主要的功能后，我很长一段时间里都没有时间去继续完善其他的东西。 比如，在我的 计划表里， 首页 是非常丰富的，可以通过 各种 内容模块 的配置来进行自由组合。

如果你是在筹划 公司级别的，需要更多功能表现，且是以内容为主的 网站， 我更推荐你去 了解 [`Astro`](https://github.com/withastro/astro) ，它也许更符合你的需求。

而 `vuepress 2.0` 到目前为止一直没有发布 正式的版本，其API变动还挺大的，暂时来说是不推荐用于 企业类型的项目中。

如果你有 对这个主题的建议，也欢迎提出，然后可以通过 PR 的方式 合并到这个项目中。

**Reply by Jetive** [reply]
非常感谢如此用心的回复。我会去了解vitepress和astro，刚刚看了astro的介绍，貌似打开了一扇全新的大门。
